site/content/in-dev/unreleased/evolution.md

```diff
@@ -0,0 +1,121 @@
+
+Polaris servers will support deprecated API endpoints / parameters / versions / etc. 
+for some transition period to allow clients to migrate.
+
+### Managing Polaris Database
+
+Polaris stores its data in a database, which is sometimes referred to as "Metastore" or
+"Persistence" in other docs.
+
+Each Polaris release may support multiple Persistence [implementations](../metastores),
+for example, "EclipseLink" (deprecated) and "JDBC" (current).
+
+Each type of Persistence evolves individually. Within each Persistence type, Polaris
+attempts to support rolling upgrades (both version X and X + 1 servers running at the
+same time).
+
+However, migrating between different Persistence types is not supported in a rolling
+upgrade manner (for example, migrating from "EclipseLink" to "JDBC"). Polaris provides
+[tools](https://github.com/apache/polaris-tools/) for migrating between different
+catalogs and those tools may be used to migrate between different Persistence types
+as well. Service interruption (downtime) should be expected in those cases.
+
+## Using Polaris as a Build-Time Dependency
+
+Polaris produces several jars. These jars or custom builds of Polaris code may be used in
+downstream projects according to the terms of the license included into Polaris distributions.
+
+This means that major version increments should be expected often (in the SemVer sense, see
+below).
+
+The minimal version of the JRE required by Polaris code (compilation target) may be updated in
+any release. Different Polaris jars may have different minimal JRE version requirements.
+
+This approach is not meant to discourage the use of Polaris code in downstream projects, but
+to allow more flexibility in evolving the codebase to support new catalog-level features
+and improve code efficiency. Maintainers of downstream projects are encouraged to join Polaris 
+mailing lists to monitor project changes, suggest improvements, and engage with the Polaris
+community in case of specific compatibility concerns.   
+
+## Semantic Versioning
+
+Polaris strives to follow [Semantic Versioning](https://semver.org/) conventions both with
+respect to Java code and REST APIs.
+
+The API surface for the purpose of Semantic Versioning is defined as follows.
+
+**Client-Facing API**
+
+* Iceberg REST Catalog API and Generic Tables API (refer to this [link](../polaris-catalog-service/)
+for their combined Open API definition).
+  * Note: Polaris implementing an optional Iceberg REST Catalog feature that was unimplemented
+  in the previous release is not considered a major change.
+  * Supporting a new revision of the Iceberg REST Catalog spec in a backward-compatible way
+  is not considered a major change.
+  * Changing the implementation of an Iceberg REST Catalog feature / endpoint in a non-backward
+  compatible way is a major change.
+* [Polaris Management API](../polaris-management-service/)
+* [Polaris Policies](http://localhost:1313/in-dev/unreleased/policy/)
```

Comment:
Again, as I said here, https://github.com/apache/polaris/pull/1890#discussion_r2150806575. I don't think the REST APIs follow Semver. Can we remove them here?

Reply:
I'm not sure I follow your suggestion, @flyrain . This is not about identifying a particular API spec revision (v1, v2, etc.) but about changes in Polaris server behaviour.

IMHO, adding support for a new API spec version (e.g. v2) is not a major change from the SemVer perspective, because it does not break backward-compatibility. In that regard, Polaris may support several API spec versions in the same release of Polaris.

On the other hand, if Polaris has a bug in implementing the Catalog REST API (for example) such that fixing it introduced a backward incompatible change in behaviour, why would we not do a major Polaris version bump? I think it would make sense. WDYT?